internal/acceptance/openstack/networking/v2/extensions/bgpvpns/bgpvpns_test.go

```diff
@@ -0,0 +1,247 @@
+//go:build acceptance || networking || bgp || bgpvpns
+
+package bgpvpns
+
+import (
+	"context"
+	"testing"
+
+	"github.com/gophercloud/gophercloud/v2/internal/acceptance/clients"
+	networking "github.com/gophercloud/gophercloud/v2/internal/acceptance/openstack/networking/v2"
+	"github.com/gophercloud/gophercloud/v2/internal/acceptance/openstack/networking/v2/extensions/layer3"
+	"github.com/gophercloud/gophercloud/v2/internal/acceptance/tools"
+	"github.com/gophercloud/gophercloud/v2/openstack/networking/v2/extensions/bgpvpns"
+	th "github.com/gophercloud/gophercloud/v2/testhelper"
+)
+
+func TestBGPVPNCRUD(t *testing.T) {
+	clients.RequireAdmin(t)
+
+	client, err := clients.NewNetworkV2Client()
+	th.AssertNoErr(t, err)
+
+	// Create a BGP VPN
+	bgpVpnCreated, err := CreateBGPVPN(t, client)
+	th.AssertNoErr(t, err)
+
+	// Get a BGP VPN
+	bgpVpnGot, err := bgpvpns.Get(context.TODO(), client, bgpVpnCreated.ID).Extract()
+	th.AssertNoErr(t, err)
+	th.AssertEquals(t, bgpVpnCreated.ID, bgpVpnGot.ID)
+	th.AssertEquals(t, bgpVpnCreated.Name, bgpVpnGot.Name)
+
+	// Update a BGP VPN
+	newBGPVPNName := tools.RandomString("TESTACC-BGPVPN-", 10)
+	updateBGPOpts := bgpvpns.UpdateOpts{
+		Name: &newBGPVPNName,
+	}
+	bgpVpnUpdated, err := bgpvpns.Update(context.TODO(), client, bgpVpnGot.ID, updateBGPOpts).Extract()
+	th.AssertNoErr(t, err)
+	th.AssertEquals(t, bgpVpnUpdated.Name, newBGPVPNName)
```

Comment:
The arguments should be reversed. It's `expected, actual`.

Reply:
You have the same issue in a few places.

Reply:
true, I always mix them up. Will fix soon.